Add starter packs to resources page

samderanova opened the pull request:
- Update navigation bar with new route to resources
- Add icons for each type of starter pack, following the design from graphics

Commit:
Add starter packs to resources page

samderanova commented on src/assets/data/resources.json:
```diff
@@ -0,0 +1,184 @@
+			},
+			{
+				"name": "Pytorch",
+				"link": "https://pytorch.org/",
+				"tooltip": "A machine learning framework."
+			},
+			{
+				"name": "Scikit-learn",
+				"link": "https://scikit-learn.org/stable/",
+				"tooltip": "A data analysis and machine learning library."
+			}
+		]
+	},
+	{
+		"name": "General Resources",
+		"description": "Gotta build your projects with something!",
+		"resources": [
+			{
+				"name": "Git",
+				"link": "https://docs.google.com/presentation/d/1Omd_azmTLQqrIjc6Bkva7PetFeQosLpfyLox8xDA5Mg/edit",
+				"tooltip": "A popular version control tool!"
+			},
+			{
+				"name": "VS Code",
+				"link": "https://code.visualstudio.com/",
+				"tooltip": "A streamlined code editor with support for debugging, task running, and version control."
+			},
+			{
+				"name": "Postman",
+				"link": "https://www.getpostman.com/",
+				"tooltip": "Use Postman to test APIs!"
+			},
+			{
+				"name": "Ngrok",
+				"link": "https://ngrok.io",
+				"tooltip": "Instantly create a public URL for a website running locally on your development machine."
+			}
+		]
+	}
+]
```
Uhh anyone have any idea what this means?

rileysw replied:
The two files are the same on my end 😩

taesungh replied:
Could be your line endings, i.e. CRLF vs LF

Commit:
Remove unnecessary div and add Title for resources page

rileysw committed:
added google cloud credits to resources

taesungh commented on src/views/resources/Resource/Resource.tsx:
```diff
@@ -0,0 +1,39 @@
+				overlay={<Tooltip id={"resource-description"}>{tooltip}</Tooltip>}
+			>
+				<a
+					className={
+						(resourceIndex % 2 === 0
+							? styles["badge-gold"]
+							: styles["badge-dark"]) + " d-block my-3 p-3 rounded-pill badge"
+					}
+					href={link}
+					style={{ textDecoration: "none" }}
```
Thought [non-blocking]: using button styles might make this simpler, but I don't want to think about it right now.

taesungh commented on src/views/resources/Resources.tsx:
```diff
@@ -0,0 +1,69 @@
+import Head from "next/head";
+import resources from "assets/data/resources.json";
+import aiML from "assets/icons/ai-ml.svg";
+import backend from "assets/icons/backend.svg";
+import database from "assets/icons/database-fill.svg";
+import deploy from "assets/icons/deploy.svg";
+import userInterface from "assets/icons/display.svg";
+import gaming from "assets/icons/gaming.svg";
+import tools from "assets/icons/tools.svg";
```
Note: for future reference, might be nice to include the source of these icon assets as part of the commit message body.

taesungh reviewed (changes requested):
Looks mostly good, small issue on the hover transition.

rileysw committed:
added render.com as resource

RandyHuynh5815 commented on src/views/resources/Resource/Resource.tsx:
```diff
@@ -0,0 +1,39 @@
+}
+
+function Resource({ resourceIndex, name, link, tooltip }: ResourceProps) {
+	const target = useRef(null);
```
It's used in the a element within the OverlayTrigger.

samderanova replied:
^^

RandyHuynh5815 committed:
Cleaned resource transition and mapping

taesungh commented on src/views/resources/Resource/Resource.module.scss:
```diff
@@ -0,0 +1,22 @@
+
+	&:hover {
+		background-color: $light-orange;
+		transition: 0.25s ease;
```
@RandyHuynh5815, we should also constrain the transition property

lassenordahl commented:
This site is lovely! I have one request. Could you update the CockroachDB link to:

https://www.cockroachlabs.com/docs?utm_source=hackuci2023&utm_medium=event&utm_campaign=hackathon

so we could do some cross domain tracking for Hackathon users? That would be awesome.

lassenordahl committed:
ui: add utm parameters for the cockroachdb docs

This link will help us track CockroachDB users across our docs and cloud
using our marketing links. Really appreciate it!

Release note: None

Commit:
Update transition property

Commit:
Remove use of ref in Resource.tsx

taesungh commented on src/views/resources/Resources.tsx:
```diff
@@ -0,0 +1,83 @@
+
+import { TitleBanner } from "components";
+
+interface IndexMapValue {
+	svg: any;
+	color: string;
+}
+
+interface IndexMap {
+	[key: string]: IndexMapValue;
+}
+
+const RESOURCE_INDEX_MAP: IndexMap = {
+	"User Interface": { svg: userInterface, color: "gold" } as IndexMapValue,
+	"Backend": { svg: backend, color: "blue" } as IndexMapValue,
+	"Databases": { svg: database, color: "gold" } as IndexMapValue,
+	"Deployment": { svg: deploy, color: "blue" } as IndexMapValue,
+	"Gaming": { svg: gaming, color: "gold" } as IndexMapValue,
+	"AI / Machine Learning": { svg: aiML, color: "blue" } as IndexMapValue,
+	"General Resources": { svg: tools, color: "gold" } as IndexMapValue,
+};
```
@RandyHuynh5815, this is a bit more complicated than I was expecting, but it will work for now.

taesungh reviewed (dismissed):
Checks workflow didn't run since there are merge conflicts, but functionality alright :/

njhuey commented on src/views/resources/Resource/Resource.tsx:
```diff
@@ -0,0 +1,36 @@
+import { useRef } from "react";
```
Unused import

samderanova committed:
Merge branch 'main' into setup/resources

samderanova committed:
Update src/views/resources/Resources.tsx

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
